chain-signatures/contract/src/lib.rs

```diff
@@ -634,6 +640,22 @@ impl VersionedMpcContract {
         }))
     }
 
+    /// This will be called internally by the contract to migrate the state when a new contract
+    /// is deployed. This function should be changed every time state is changed to do the proper
+    /// migrate flow.
+    ///
+    /// If nothing is changed, then this function will just return the current state. If it fails
+    /// to read the state, then it will return an error.
+    #[private]
+    #[init(ignore_state)]
+    #[handle_result]
+    pub fn migrate() -> Result<Self, MpcContractError> {
```

Comment:
We must keep the last deployed version somewhere to ensure migrations are successful.

Reply:
yeah, I think we should have a list of structs when we're making these breaking changes